File: python/design-patterns/state/context/player.py
class Player:
    def __init__(self): self.state = Stopped()
    def set_state(self, s): self.state = s
    def play(self):  self.state.play(self)
    def pause(self): self.state.pause(self)
    def stop(self):  self.state.stop(self)

File: java/design-patterns/state/naive/Player.java
class Player {
    private boolean isPlaying = false;
    private boolean isPaused  = false;

    void play() {
        if (isPlaying) return;                   // already playing
        isPaused = false;
        isPlaying = true;                        // stopped/paused -> play
    }
    void pause() {
        if (!isPlaying) return;                  // ❌ can't pause if not playing
        isPlaying = false; isPaused = true;
    }
    void stop() {
        // ❌ every method re-checks the same flags
        isPlaying = false; isPaused = false;
    }
}